assets/dev/js/editor/views/base-container.js

```diff
@@ -88,6 +88,80 @@ module.exports = Marionette.CompositeView.extend( {
 		return newView;
 	},
 
+	createElementFromContainer( container, options = {} ) {
+		const model = container.model.attributes;
+
+		return this.createElementFromModel( Object.assign(
+			model,
+			model.custom,
+			{ settings: model.settings.attributes }
+		), options );
+	},
+
+	createElementFromModel( model, options = {} ) {
+		if ( model instanceof Backbone.Model ) {
+			model = model.attributes;
+		}
+
+		model = Object.assign( model, model.custom );
+
+		if ( elementor.helpers.maybeDisableWidget( model ) ) {
+			return;
+		}
+
+		const historyId = $e.internal( 'document/history/start-log', {
+			type: this.getHistoryType( options.event ),
+			title: elementor.helpers.getModelLabel( model ),
+		} );
+		let container = this.getContainer();
+
+		if ( options.newSection ) {
+			container = $e.run( 'document/elements/create', {
+				model: {
+					elType: 'section',
```

Comment:
Consult with @iNewLegend

Reply:
@matipojo I think that each view should have `createElement` method which know how to handle the situation for each view, **preview**: should know add section, **section**: should know add column etc...
`onDropping` can call to `createElement`
`browser-import` can call to `createElement`